mlflow/utils/databricks_utils.py

```diff
@@ -419,28 +430,36 @@ def _fail_malformed_databricks_auth(profile):
     )
 
 
-def get_databricks_host_creds(server_uri=None):
-    """
-    Reads in configuration necessary to make HTTP requests to a Databricks server. This
-    uses the Databricks CLI's ConfigProvider interface to load the DatabricksConfig object.
-    If no Databricks CLI profile is found corresponding to the server URI, this function
-    will attempt to retrieve these credentials from the Databricks Secret Manager. For that to work,
-    the server URI will need to be of the following format: "databricks://scope:prefix". In the
-    Databricks Secret Manager, we will query for a secret in the scope "<scope>" for secrets with
-    keys of the form "<prefix>-host" and "<prefix>-token". Note that this prefix *cannot* be empty
-    if trying to authenticate with this method. If found, those host credentials will be used. This
-    method will throw an exception if sufficient auth cannot be found.
+def _fail_model_serving_creds_env(exception):
+    raise MlflowException(
+        "Unable to read Oauth credentials from file mount for Databricks "
+        "Model Serving dependency failed"
+    ) from exception
+
+
+# constant defined outside function for testing override
+_MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH = "/var/credentials-secret/model-dependencies-oauth-token"
+
+# Helper function to attempt to read OAuth Token from
+# mounted file in Databricks Model Serving environment
+def _get_model_dependency_oauth_token(should_retry=True):
+    try:
+        with open(_MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH) as f:
+            oauth_dict = json.load(f)
+            return oauth_dict["OAUTH_TOKEN"][0]["oauthTokenValue"]
+    except Exception as e:
+        # sleep and retry in case of any race conditions with OAuth refreshing
+        if should_retry:
+            time.sleep(0.1)
+            return _get_model_dependency_oauth_token(should_retry=False)
+        else:
+            _fail_model_serving_creds_env(e)
 
-    Args:
-        server_uri: A URI that specifies the Databricks profile you want to use for making
-            requests.
 
-    Returns:
-        MlflowHostCreds which includes the hostname and authentication information necessary to
-        talk to the Databricks server.
-    """
+def _default_databricks_host_creds(server_uri):
     profile, path = get_db_info_from_uri(server_uri)
     config = ProfileConfigProvider(profile).get_config() if profile else get_config()
+    insecure = hasattr(config, "insecure") and config.insecure
```

Comment:
```suggestion
    insecure = getattr(config, "insecure", False)
```

can we use `getattr`?